### DISCUSSION TITLE
proposal: Remove github engine

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#7286] [Ideas]
Nearly no one uses the GitHub engines. But we need to maintain it everywhere in our code base. Since the `FUSE` engine is much more mature now, we don't need a toy engine for demonstration. 

So I propose to remove the GitHub engine entirely.

---

Tracked at https://github.com/datafuselabs/databend/issues/7287

**Comment by bohutang** [comment]
Agree +1

**Comment by sundy-li** [comment]
Ok,  let's avoid adding too many useless but cool features in the main branch.
